app/models/event.rb

```diff
@@ -66,6 +66,17 @@ def search_activities(author = nil, query_string = "", filter = "all")
   end
   alias_method :activities, :search_activities
 
+  def activities_per_day(author = nil, query_string = "", filter = "all")
+    activities = search_activities(author, query_string, filter)
+    activities.sort_by! { |activity| activity.dates.first }
+
+    activities.each_with_object(Hash.new { [] }) do |activity, grouped_by_day|
```

Comment:
But I'm confused... If I try `Hash.new { [] }` in IRB, I just get an empty hash. Why not just use `{}`?